src/modules/godot.rs

```diff
@@ -0,0 +1,169 @@
+use crate::configs::godot::GodotConfig;
+
+use super::{Context, Module};
+
+/// Creates a module that tells if the current directory is in a Godot project.
+pub fn module<'a>(context: &'a Context) -> Option<Module<'a>> {
+    use super::ModuleConfig;
+    use crate::formatter::StringFormatter;
+    use crate::formatter::VersionFormatter;
+
+    let godot_project_root = context
+        .begin_ancestor_scan()
+        .set_files(&["project.godot"])
+        .scan();
+    godot_project_root?;
+
+    let mut module = context.new_module("godot");
+    let config: GodotConfig = GodotConfig::try_load(module.config);
+
+    let parsed = StringFormatter::new(config.format).and_then(|formatter| {
+        formatter
+            .map_meta(|variable, _| match variable {
+                "symbol" => Some(config.symbol),
+                _ => None,
+            })
+            .map_style(|variable| match variable {
+                "style" => Some(Ok(config.style)),
+                _ => None,
+            })
+            .map(|variable| match variable {
+                "version" => match config.show_version {
+                    true => {
+                        let version_string = get_godot_version(context, &config)?;
+                        VersionFormatter::format_module_version(
+                            module.get_name(),
+                            &version_string,
+                            config.version_format,
+                        )
+                        .map(Ok)
+                    }
+                    false => None,
+                },
+                _ => None,
+            })
+            .parse(None, Some(context))
+    });
+
+    module.set_segments(match parsed {
+        Ok(segments) => segments,
+        Err(error) => {
+            log::warn!("Error in module `godot`:\n{}", error);
+            return None;
+        }
+    });
+
+    Some(module)
+}
+
+/// Gets the godot version using the configured command.
+fn get_godot_version(context: &Context, config: &GodotConfig) -> Option<String> {
+    let version_output = context
+        .exec_cmd(config.version_command[0], &config.version_command[1..])?
+        .stdout;
+    let mut period_count = 0;
+    for (index, char) in version_output.char_indices() {
+        if char == '.' {
+            period_count += 1;
+            if period_count == 3 {
+                let (version_string, _) = version_output.split_at(index);
+                return Some(String::from(version_string));
+            }
+        }
+    }
+    None
```

Comment:
I would prefer a regex or `split` & `take_while` here. Instead of only counting dots, please also make sure the text in-between is only compromised of digits.

Reply:
I pushed an update that uses `split`, `take`, and `take_while` here, and I also updated the unit tests to more thoroughly cover this function.